apis/v1beta1/gateway_types.go

```diff
@@ -148,6 +154,72 @@ type GatewaySpec struct {
 	Addresses []GatewayAddress `json:"addresses,omitempty"`
 }
 
+// GatewayInfrastructure defines infrastructure level attributes about a Gateway
+type GatewayInfrastructure struct {
+	// Routability allows the Gateway to specify the accessibility of its addresses. Setting
+	// this property will override the default value defined by the GatewayClass.
+	//
+	// If the desired Gateway routability is incompatible with the GatewayClass implementations
+	// MUST set the condition `Accepted` to `False` with `Reason` set to `UnsupportedRoutability`.
+	//
+	// The default value of routability is implementation specific and MUST remain consistent for
+	// Gateways with the same gatewayClassName
+	//
+	// Implementations MAY leave this property unset. The routability MUST appear in the GatewayStatusAddress.
```

Comment:
This part still confuses me, this type is only in spec right? Implementations should _never_ be setting values in spec. That routability must be set in status feels like it should be stated on status, not here.

Reply:
> This part still confuses me, this type is only in spec right? 

Yes it's only in the spec.

>  Implementations should never be setting values in spec. 

The implementation could apply defaults - similar to how K8s defaults pull policy etc.

Reply:
> That routability must be set in status feels like it should be stated on status, not here.

I'll drop this this from here. In the GatewayStatusAddress we have

> Implementations that support Routability MUST populate this field

Reply:
>implementation could apply defaults

Agree that they could, but if they do, I think it should be via status, not spec (unless it's something like a defaulting webhook that the user intentionally installs, but I don't think that's really something we need to mention in our API spec).

Reply:
Ok - sounds good I'll drop this.

Reply:
done